core/src/main/java/io/grpc/ServerServiceDefinition.java

```diff
@@ -126,6 +126,21 @@ private Builder(ServiceDescriptor serviceDescriptor) {
      * Construct new ServerServiceDefinition.
      */
     public ServerServiceDefinition build() {
+      Map<String, ServerMethodDefinition<?, ?>> tmpMethods =
+          new HashMap<String, ServerMethodDefinition<?, ?>>(methods);
+      for (MethodDescriptor<?, ?> descriptorMethod : serviceDescriptor.getMethods()) {
+        ServerMethodDefinition<?, ?> removed = tmpMethods.remove(
```

Comment:
Lets start with instance equality and we can adjust later if someone has a real need for something else.